### PR TITLE
[Core] [Mono] Transform2D rotation and scale fixes

### PR DESCRIPTION
**Pull request by aaronfranke**
Core:

* Add `set_scale` method to Transform2D. Not exposed to GDScript yet, no user-facing changes.

* Fix `set_rotation` to work with scaled transforms. Also not exposed to GDScript.

* Use `SGN()` method instead of manually writing `> 0 ? 1 : -1`.

Mono: @neikeq 

* Fix rotation and scale getters.

* Add rotation and scale setters.

* Some internal code refactoring.

### CODE REVIEW

File: core/math/transform_2d.cpp [hunk 1]
```diff
@@ -80,13 +80,14 @@ real_t Transform2D::get_rotation() const {
...
 void Transform2D::set_rotation(real_t p_rot) {
-
+	Vector2 scale = get_scale();
 	real_t cr = Math::cos(p_rot);
 	real_t sr = Math::sin(p_rot);
 	elements[0][0] = cr;
 	elements[0][1] = sr;
 	elements[1][0] = -sr;
 	elements[1][1] = cr;
+	set_scale(scale);
```
Comment (akien-mga) [inline core/math/transform_2d.cpp]: What's the rationale for this change? It breaks compat, so it would be good to justify it a bit :)

Comment (aaronfranke) [reply core/math/transform_2d.cpp]: With the old code, the existing basis values were not used, such that it always ends up with a scale of 1. This change makes it so that it takes the previous scale and applies it after setting rotation. Transforms that already had a scale of 1 shouldn't be affected.

If there is a desire for a method that sets rotation and deleting the existing scale, I could add that (and possibly change the existing calls to that) but it probably isn't a good idea for a method simply called `set_rotation` to delete the scale.

Comment (akien-mga) [reply core/math/transform_2d.cpp]: Thanks for the explanation. @reduz confirmed that this is a bug worth fixing for 3.1 :)

**Review by akien-mga — APPROVED** [review]
Ok for the `core` changes.

**Review by neikeq — APPROVED** [review]

**Comment by YeldhamDev** [comment]
Is there a reason why those aren't exposed to GDScript?

**Comment by aaronfranke** [comment]
It just hasn't been done yet, I don't see any reason for them not to be. I could add that, but I'm not sure if this is a bad time due to feature freeze (better to wait for 3.2?).

**Comment by YeldhamDev** [comment]
Considering that it's just exposing something that's already there, I don't see the problem.
